spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala

```diff
@@ -91,6 +98,30 @@ class CometSparkSessionExtensions
               scanExec.copy(scan = cometScan),
               runtimeFilters = scanExec.runtimeFilters)
 
+          // unsupported parquet data source V2
+          case scanExec: BatchScanExec if scanExec.scan.isInstanceOf[ParquetScan] =>
+            val requiredSchema = scanExec.scan.asInstanceOf[ParquetScan].readDataSchema
+            var info1: Option[String] = None
+            if (isSchemaSupported(requiredSchema)) {
+              info1 = Some(s"Schema $requiredSchema is not supported")
+            }
```

Comment:
I wonder if we could add a helper method to reduce the boilerplate for each of these sections for creating the `Option[String]`.

We could add a method like this:

```scala
def createMessage(condition: Boolean, message: () => String): Option[String] = {
  if (condition) {
    Some(message())
  } else {
    None
  }
}
```

The call site could then be a one-liner:

```scala
val info1 = createMessage(isSchemaSupported(requiredSchema), s"Schema $requiredSchema is not supported")
```

What do you think?

Reply:
edit: I just updated the suggested code to use the pass-by-name `message: => String` syntax so that the message is evaluated lazily

Reply:
Done.